File: testcases/benchmark/recursion/tower_of_hanoi.cpp
#include <iostream>

int move_count = 0;

void tower_of_hanoi(int n, char from, char to, char aux) {
    if (n == 1) {
        move_count++;
        return;
    }
    tower_of_hanoi(n - 1, from, aux, to);
    move_count++;
    tower_of_hanoi(n - 1, aux, to, from);
}

int main() { return 0; }

File: testcases/benchmark/recursion/tower_of_hanoi.rs
static mut MOVE_COUNT: i32 = 0;

fn tower_of_hanoi(n: i32, _from: char, _to: char, _aux: char) {
    unsafe {
        if n == 1 {
            MOVE_COUNT += 1;
            return;
        }
        tower_of_hanoi(n - 1, _from, _aux, _to);
        MOVE_COUNT += 1;
        tower_of_hanoi(n - 1, _aux, _to, _from);
    }
}

fn main() {}
